src/Maestro/Microsoft.DotNet.Maestro.Tasks/src/PushMetadataToBuildAssetRegistry.cs

```diff
@@ -495,5 +578,186 @@ private void LookupForMatchingGitHubRepository(BuildData mergedBuild)
                 }
             }
         }
+
+        /// <summary>
+        /// Creates an AssetData object for the merged manifest so it can be injected
+        /// in itself
+        /// </summary>
+        /// <param name="assets">List of assets extracted from the merged manifest</param>
+        /// <param name="location">Initial location for the merged manifest entry</param>
+        /// <param name="manifestFileName">Merged manifest file name</param>
+        /// <returns>An AssetData with data about the merge manifest</returns>
+        private AssetData GetManifestAsAsset(IImmutableList<AssetData> assets, string location, string manifestFileName)
+        {
+            (string accountName, string projectName, string azDORepoName) = AzureDevOpsClient.ParseRepoUri(GetAzDevRepository());
+
+            if (string.IsNullOrEmpty(AssetVersion))
+            {
+                AssetData asset = assets.Where(a => a.NonShipping).FirstOrDefault();
+
+                if (asset != null)
+                {
+                    AssetVersion = asset.Version;
+                }
+            }
+
+            AssetData assetData = new AssetData(true)
+            {
+                Locations = (location == null) ? null : ImmutableList.Create(new AssetLocationData(LocationType.Container)
```

Comment:
Is a null location here handled correctly everywhere? Is it safer to return an empty list?

Reply:
`AssetData` is what is POSTed to BAR and null (for empty) is expected.

```json
{
      "id": 4782687,
      "name": "assets/symbols/Microsoft.DotNet.AsmDiff.5.0.0-beta.20351.8.symbols.nupkg",
      "version": "5.0.0-beta.20351.8",
      "buildId": 56549,
      "nonShipping": true,
      "locations": null
    },
```